### PR TITLE
Refactorization of PingListener

### PR DESCRIPTION
**Pull request by SzinDzeks**

### CODE REVIEW

File: src/main/java/me/axerr/pingwheel/ping/PingListener.java [hunk 1]
```diff
@@ -17,55 +18,91 @@ public class PingListener implements PluginMessageListener {
...
     }
 
-    private boolean validateRateLimit(Player player) {
-        boolean canPing = !PingWheel.getRateLimiter().canPing(player) && !PingWheel.getRateLimiter().checkBypass(player);
-        if (!canPing)
-            sendMessage(player, Config.RATE_LIMIT_MESSAGE);
-        return canPing;
+    private boolean doesPlayerHavePermissionToUsePing(Ping ping) {
+        Player player = ping.getPlayer();
+        if(ping.isEntity())
```
Comment (SandResidence) [inline src/main/java/me/axerr/pingwheel/ping/PingListener.java:86]: Add space pls

File: src/main/java/me/axerr/pingwheel/ping/PingListener.java [hunk 2]
```diff
@@ -17,55 +18,91 @@ public class PingListener implements PluginMessageListener {
...
-            sendMessage(player, Config.ENTITY_PING_NO_PERMISSION_MESSAGE);
-        return valid;
+
+        if(!canPlayerUsePing(ping)) return false;
+        if(!isPingWithinXYZLimits(ping)) return false;
+        if(!isInsidePingableWorldGuardRegion(ping)) {
+            sendMessage(player, Config.WORLDGUARD_DENY_MESSAGE);
+            return false;
+        }
+        if(isPlayerRateLimited(player)){
```
Comment (SandResidence) [inline src/main/java/me/axerr/pingwheel/ping/PingListener.java:38]: Add spaces between if and (

File: src/main/java/me/axerr/pingwheel/ping/PingListener.java [hunk 3]
```diff
@@ -17,55 +18,91 @@ public class PingListener implements PluginMessageListener {
     @Override
     public void onPluginMessageReceived(String channel, Player player, byte[] message) {
         FriendlyByteBuf data = new FriendlyByteBuf(Unpooled.copiedBuffer(message));
-
         Ping ping = new Ping(player, data);
 
-        if (!validateEntityPing(ping) || !validateLocationPing(ping) || !validateLimits(ping) || !validateWorldGuardFlag(ping) || !validateRateLimit(player))
-            return;
+        if(!isPingAllowed(ping)) return;
```
Comment (SandResidence) [inline src/main/java/me/axerr/pingwheel/ping/PingListener.java:23]: Add space between if and (
Add enter before return.

File: src/main/java/me/axerr/pingwheel/ping/PingListener.java [hunk 4]
```diff
@@ -17,55 +18,91 @@ public class PingListener implements PluginMessageListener {
...
+        if(isPlayerRateLimited(player)){
+            sendMessage(player, Config.RATE_LIMIT_MESSAGE);
+            return false;
+        }
+        return true;
     }
 
-    private boolean validateLocationPing(Ping ping) {
+    private boolean canPlayerUsePing(Ping ping) {
+        if(!isPermissionToUsePingRequired(ping)) return true;
```
Comment (SandResidence) [inline src/main/java/me/axerr/pingwheel/ping/PingListener.java:46]: Add space between if and ( and new line before return statement.

File: src/main/java/me/axerr/pingwheel/ping/PingListener.java [hunk 5]
```diff
@@ -17,55 +18,91 @@ public class PingListener implements PluginMessageListener {
...
+        Location location = ping.getLocation();
+        return WorldGuardHook.canLocationBePinged(location, player);
     }
 
-    private boolean validateLimits(Ping ping) {
-        return isWithinLimits(ping.getX(), Config.LIMIT_X_ENABLED, Config.LIMIT_X_MIN, Config.LIMIT_X_MAX)
-                && isWithinLimits(ping.getY(), Config.LIMIT_Y_ENABLED, Config.LIMIT_Y_MIN, Config.LIMIT_Y_MAX)
-                && isWithinLimits(ping.getZ(), Config.LIMIT_Z_ENABLED, Config.LIMIT_Z_MIN, Config.LIMIT_Z_MAX);
+    private boolean isPlayerRateLimited(Player player) {
+        if(PingWheel.getRateLimiter().checkBypass(player)) return false;
```
Comment (SandResidence) [inline src/main/java/me/axerr/pingwheel/ping/PingListener.java:65]: Same thing bro.

File: src/main/java/me/axerr/pingwheel/ping/PingListener.java [hunk 6]
```diff
@@ -17,55 +18,91 @@ public class PingListener implements PluginMessageListener {
...
+        return !PingWheel.getRateLimiter().canPing(player);
     }
 
-    private boolean isWithinLimits(double value, boolean isEnabled, double minLimit, double maxLimit) {
-        return !isEnabled || (value >= minLimit && value <= maxLimit);
+    private boolean isPermissionToUsePingRequired(Ping ping) {
+        if(ping.isEntity())
```
Comment (SandResidence) [inline src/main/java/me/axerr/pingwheel/ping/PingListener.java:70]: Here too

File: src/main/java/me/axerr/pingwheel/ping/PingListener.java [hunk 7]
```diff
@@ -17,55 +18,91 @@ public class PingListener implements PluginMessageListener {
...
     }
 
-    private boolean validateWorldGuardFlag(Ping ping) {
+    private void sendPingNoPermissionMessage(Ping ping) {
         Player player = ping.getPlayer();
-        boolean canPing = WorldGuardHook.canLocationBePinged(ping.getLocation(), player);
-        if (!canPing)
-            sendMessage(player, Config.WORLDGUARD_DENY_MESSAGE);
-        return canPing;
+        if(ping.isEntity())
```
Comment (SandResidence) [inline src/main/java/me/axerr/pingwheel/ping/PingListener.java:78]: Bro again

**Review by SandResidence — CHANGES_REQUESTED** [review]
Waiting for changes...

**Comment by SzinDzeks** [comment]
bro go fish

**Comment by SzinDzeks** [comment]
sry bro pls merge

**Pull request merged**